### PR TITLE
Join sequence of composable into attribute string

### LINKED ISSUE #328: Join array of attributed strings with a Composable
It would be nice to have this:

```swift
public extension Sequence where Element: Composable {

    public func joined(separator: Composable) {
        ...
    }

}
```

### PR DESCRIPTION
**Pull request by ZevEisenberg**
Closes #328

### CODE REVIEW

**Comment by ZevEisenberg** [comment]
@jvisenti I pushed an additional commit. Could you please have a look at it? I'm wondering whether it's confusing to have `["a", "b"].joined()` be able to produce either a `String` or an `NSAttributedString` depending on type inference. cc @KingOfBrian

File: Tests/ComposableTests.swift [hunk 1]
```diff
@@ -173,4 +173,41 @@ class ComposableTests: XCTestCase {
         XCTAssertEqual(paragraphEnd.lineSpacing, 0)
     }
 
+    func testJoiningSequence() {
+        let redStyle = StringStyle(.color(.red))
+        let greenStyle = StringStyle(.color(.red))
+        let blueStyle = StringStyle(.color(.red))
```
Comment (KingOfBrian) [inline Tests/ComposableTests.swift]: Strange shade of blue / green.

Comment (ZevEisenberg) [reply Tests/ComposableTests.swift]: d'oh, thank you

**Comment by KingOfBrian** [comment]
Good call on the oddity of having the overloads available, but I think it's fine. It may have an impact on compile times, but I doubt it's significant. It could be surprising to a user, but I'm not sure it's a big gotcha. I wonder if there's any guidance on providing overloaded methods.

**Comment by jvisenti** [comment]
I think in general the context will define the type and there won't be much ambiguity

**Comment by ZevEisenberg** [comment]
I also confirmed that, without any other context, `["a", "b"].joined()` returns a `String`, so we should be OK.

**Pull request merged**